app/src/main/java/edu/agh/susgame/front/managers/LobbyManager.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.runtime.snapshots.SnapshotStateMap
+import androidx.compose.runtime.toMutableStateList
+import edu.agh.susgame.dto.rest.model.GameMapDTO
+import edu.agh.susgame.dto.rest.model.Lobby
+import edu.agh.susgame.dto.rest.model.LobbyId
+import edu.agh.susgame.dto.rest.model.PlayerId
+import edu.agh.susgame.dto.rest.model.PlayerNickname
+import edu.agh.susgame.dto.rest.model.PlayerREST
+import edu.agh.susgame.front.gui.components.common.graph.edge.Edge
+import edu.agh.susgame.front.gui.components.common.graph.edge.EdgeId
+import edu.agh.susgame.front.gui.components.common.graph.node.Host
+import edu.agh.susgame.front.gui.components.common.graph.node.Node
+import edu.agh.susgame.front.gui.components.common.graph.node.NodeId
+import edu.agh.susgame.front.gui.components.common.graph.node.Router
+import edu.agh.susgame.front.gui.components.common.graph.node.Server
+import edu.agh.susgame.front.gui.components.common.util.Coordinates
+import edu.agh.susgame.front.gui.components.common.util.player.PlayerLobby
+import edu.agh.susgame.front.gui.components.common.util.player.PlayerStatus
+import edu.agh.susgame.front.service.interfaces.LobbyService
+import edu.agh.susgame.front.service.mock.createCustomMapState
+
+
+class LobbyManager(
+    val lobbyService: LobbyService,
+    var id: LobbyId,
+    var name: String,
+    var maxNumOfPlayers: Int,
+    var gameTime: Int,
+    var playersRest: MutableMap<PlayerId, PlayerREST> = mutableMapOf(),
+    var playersMap: SnapshotStateMap<PlayerId, PlayerLobby> = mutableStateMapOf(),
+    var gameManager: MutableState<GameManager?> = mutableStateOf(null),
+    var isGameReady: MutableState<Boolean> = mutableStateOf(false),
+    var localPlayer: PlayerLobby = PlayerLobby()
+
+) {
+    fun updateFromRest(lobby: Lobby) {
+        lobby.playersWaiting.forEach() {
+            addPlayerRest(it)
+        }
+    }
+    fun addLocalPlayer(){
+        val localPlayerRest = localPlayer.id?.let { PlayerREST(localPlayer.name, it,1234) }
+        if (localPlayerRest != null) {
+            addPlayerRest(localPlayerRest)
+        }
+    }
+
+    fun getPlayerStatus(id: PlayerId) = playersMap[id]?.status?.value
+
+    fun addPlayerRest(player: PlayerREST) {
+        playersMap[player.id] = PlayerLobby(
+            name = player.nickname,
+            status = mutableStateOf(PlayerStatus.NOT_READY)
+        )
+        playersRest[player.id] = player
+    }
+
+    fun getHowManyPlayersInLobby() = playersMap.size
+
+
+    fun updatePlayerStatus(id: PlayerId, status: PlayerStatus) {
+        playersMap[id]?.status?.value = status
+    }
+
+    fun deletePlayer(playerId: PlayerId) {
+        playersMap.remove(playerId)
+        playersRest.remove(playerId)
+    }
+
+    fun getMapFromServer() {
+        this.id.let { id ->
+            lobbyService.getGameMap(id)
+                .thenApply { result -> parseMap(result) }
+        }
+    }
+
+
+    private fun parseMap(gameMapDTO: GameMapDTO?) {
```

Comment:
Przeniósłbym większość tej logiki do jakiegoś parsera, coś typu:
```kotlin
object DtoParser {
    fun gameMapDtoToGameManager(gameMapDTO: GameMapDTO, playerId: PlayerId): GameManager {
        // ...
    }
}
```

i potem tak mniej więcej może wyglądać `getMapFromServer()`:

```kotlin
    fun getMapFromServer() {
        this.id.let { id ->
            lobbyService.getGameMap(id)
                .thenApply { gameMapDTO ->
                    val playerId = this.localPlayer.id
                    if (gameMapDTO != null && playerId != null) {
                        gameManager.value = DtoParser.gameMapDtoToGameManager(gameMapDTO, playerId)
                        isGameReady.value = true
                    } else {
                        println("Failed to update map from the server for gameMapId: $id and player: $playerId")
                    }
                }
        }
    }
 ```

Reply:
zmieniłem w PlayerLobby domyślne id, żeby uniknąć tego nullowania, bo kompilator krzyczał że to playerId może się zmienić w czasie na null, trzeba by było dodać `!!` , a raczej dobrze tego unikać, jakby się cos psuło to będziemy widzieć